### PR TITLE
Order payment

### PR DESCRIPTION
**Pull request by pedromtec**
#### What problem is this solving?

 Save payment data using `savePaymentData` method.

#### Checklist/Reminders

- [x] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
_ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: react/OrderPayment.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import React, { createContext, ReactNode, useContext } from 'react'
+import { useQuery, useMutation } from 'react-apollo'
+import { cardSessionId as CardSessionIdQuery } from 'vtex.checkout-resources/Queries'
+import {
+  savePaymentToken as SavePaymentTokenMutation,
+  saveCards as SaveCardsMutation,
+} from 'vtex.checkout-resources/Mutations'
```
Comment (marcoskwkm) [inline react/OrderPayment.tsx]:
```suggestion
import SavePaymentTokenMutation from 'vtex.checkout-resources/MutationSavePaymentToken'
import SaveCardsMutation from 'vtex.checkout-resources/MutationSaveCards'
```
Context: https://github.com/vtex-apps/checkout-resources/pull/22

File: react/typings/vtex.checkout-resources.d.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+declare module 'vtex.checkout-resources/Queries'
+declare module 'vtex.checkout-resources/Mutations'
```
Comment (marcoskwkm) [inline react/typings/vtex.checkout-resources.d.ts]: I think if you run `vtex setup` you shouldn't need this.

File: react/__tests__/index.test.tsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import React from 'react'
```
Comment (marcoskwkm) [inline react/__tests__/index.test.tsx]: Remove this file.

File: react/components/TranslatedTitle.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { FormattedMessage } from 'react-intl'
```
Comment (marcoskwkm) [inline react/components/TranslatedTitle.tsx]: Remove this file.

**Review by marcoskwkm — CHANGES_REQUESTED** [review]

File: react/OrderPayment.tsx (hunk 1 above)
Comment (pedromtec) [reply react/OrderPayment.tsx]: I think it is a major change.

Comment (marcoskwkm) [reply react/OrderPayment.tsx]: For what reason? These are not breaking changes.

Comment (pedromtec) [reply react/OrderPayment.tsx]: Ohh sorry, the other way still works.

File: react/OrderPayment.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  paymentSystem: string
+  cardHolder: string
+  cardNumber: string
+  expiryDate: string
+  csc: string
+  document: string
+  documentType: string
+  partnerId: string
+  address: Address
+}
```
Comment (darlenedms) [inline react/OrderPayment.tsx:34]: Add an empty line between interfaces definitions.

File: react/OrderPayment.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+interface PaymentToken {
+  creditCardToken: string
+  paymentSystem: string
+}
+
+interface Status {
+  error: boolean
+  value: PaymentToken[] | string
+}
```
Comment (darlenedms) [inline react/OrderPayment.tsx:53]: Add an empty line between interfaces definitions.

File: react/OrderPayment.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+  const [saveCardsMutation] = useMutation(SaveCardsMutation)
+  const [savePaymentTokenMutation] = useMutation(SavePaymentTokenMutation)
+
+  const savePaymentData = async (
+    paymentData: PaymentData[]
+  ): Promise<Status> => {
+    try {
+      const {
+        data: { saveCards },
```
Comment (darlenedms) [inline react/OrderPayment.tsx]: Shouldn't it be save**d**Cards? 🤔 What is it exactly?

**Review by darlenedms — CHANGES_REQUESTED** [review]

File: react/OrderPayment.tsx (hunk 7 above)
Comment (pedromtec) [reply react/OrderPayment.tsx]: The mutation return looks like: 
```
data: {
  saveCards: {
    value
  }
}
```

File: react/OrderPayment.tsx [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+const getPaymentTokens = (tokenizedCards: TokenizedCard[]): PaymentToken[] =>
+  tokenizedCards.map(tokenizedCard => ({
+    creditCardToken: tokenizedCard.token,
+    paymentSystem: `${tokenizedCard.paymentSystem}`,
+  }))
+
+const OrderPaymentContext = createContext<Context | undefined>(undefined)
+
+export const OrderPaymentProvider = ({ children }: OrderPaymentProps) => {
```
Comment (lucasecdb) [inline react/OrderPayment.tsx]:
```suggestion
export const OrderPaymentProvider: React.FC<OrderPaymentProps> = ({ children }) => {
```

File: react/OrderPayment.tsx [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+      refetch()
+
+      return {
+        error: true,
+        value: 'apiError',
+      }
+    }
+  }
+  return (
+    <OrderPaymentContext.Provider value={{ savePaymentData }}>
```
Comment (lucasecdb) [inline react/OrderPayment.tsx]: you should wrap the context value within a `useMemo`, else this wil cause all childs that consume this context to re-render when this component's parent renders.

something like

```tsx
const value = useMemo(() => ({ savePaymentData }), [savePaymentData])

return (
  <OrderPaymentContext.Provider value={value}>
    {children}
  </OrderPaymentContext.Provider>
)
```

**Review by lucasecdb — APPROVED** [review]
LGTM!

File: react/package.json [hunk 10]
```diff
@@ -8,6 +8,7 @@
     "classnames": "^2.2.6",
     "ramda": "^0.26.1",
```
Comment (jeffersontuc) [inline react/package.json]: Are you using `ramda`? If not, I think might be worth removing it.

**Review by jeffersontuc — APPROVED** [review]

File: manifest.json [hunk 11]
```diff
@@ -1,12 +1,11 @@
 {
   "vendor": "vtex",
-  "name": "base-store-component",
+  "name": "order-payment",
   "version": "0.2.2",
```
Comment (marcoskwkm) [inline manifest.json]: You might want to change this to `0.0.0`.

File: README.md [hunk 12]
```diff
@@ -0,0 +1 @@
+# OrderPayment
```
Comment (marcoskwkm) [inline README.md]: Move this file to `/docs/README.md` so docs builder can read it.

**Review by marcoskwkm — CHANGES_REQUESTED** [review]
LGTM, just a small issue.

**Review by marcoskwkm — APPROVED** [review]

**Review by darlenedms — APPROVED** [review]

**Pull request merged**